firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -279,7 +279,31 @@ uint8_t TCMD_parse_full_telecommand(const char tcmd_str[], TCMD_TelecommandChann
         return 90;
     }
 
-    // Reached the end of the telecommand parsing. Thus, success. Fill the output struct.
+    // ensure the corrent number of args are provided
+    int32_t num_args_expected = TCMD_telecommand_definitions[tcmd_idx].number_of_args;
+    char* tcmd_name = TCMD_telecommand_definitions[tcmd_idx].tcmd_name;
+
+    char error_message[130];
+    snprintf(error_message, sizeof(error_message), "Error: TCMD_parse_full_telecommand: %s() accepts %d argument(s).\n", tcmd_name, num_args_expected);
+    // if ( num_args_expected == 0 && arg_len > 0) {
+    //     DEBUG_uart_print_str(error_message);
+    //     return 100;
+    // }
+    int32_t num_commas = 0;
+    for (int32_t i = 0; i < arg_len + 1; i++) {
+        if (args_str_no_parens[i] == ',') {
+            num_commas++;
+        }    
+    }
+    int32_t takes_no_arguments_and_none_provided = num_args_expected == 0 && arg_len == 0;
```

Comment:
Pretty sure that the validation logic should just be:
```
    (num_args_expected == 0 && arg_len == 0) || (num_commas == (num_args_expected-1))
```

Thoughts? Am I missing something?

Reply:
If a telecommand accepts 1 argument but no argument is given this validation logic would fail. Changing it to: `(num_args_expected == 0 && arg_len == 0) || (num_commas == (num_args_expected-1) && arg_len != 0)` should work though.

Reply:
Ah, great catch! I like that though; the three-part logic seemed to have overlapping cases, I think.